api/src/main/java/org/eclipse/microprofile/health/HealthCheckResponseBuilder.java

```diff
@@ -29,21 +29,22 @@
  * The ResponseBuilder class is reserved for an extension by implementation providers.
  * </p>
  */
-public abstract class ResponseBuilder {
+public abstract class HealthCheckResponseBuilder {
 
-    public abstract ResponseBuilder name(String name);
+    public abstract HealthCheckResponseBuilder name(String name);
 
-    public abstract ResponseBuilder withAttribute(String key, String value);
+    public abstract HealthCheckResponseBuilder withAttribute(String key, String value);
 
-    public abstract ResponseBuilder withAttribute(String key, long value);
+    public abstract HealthCheckResponseBuilder withAttribute(String key, long value);
 
-    public abstract ResponseBuilder withAttribute(String key, boolean value);
+    public abstract HealthCheckResponseBuilder withAttribute(String key, boolean value);
 
-    public abstract Response up();
+    public abstract HealthCheckResponseBuilder up();
 
-    public abstract Response down();
+    public abstract HealthCheckResponseBuilder down();
+    public abstract HealthCheckResponseBuilder state(boolean isUp);
 
-    public abstract Response state(boolean up);
```

Comment:
We might even extend to support `BooleanSupplier` as in : 

```
HealthCheckResponseBuilder.state(BooleanSupplier sup);
```

which would allow for 

```
state( () -> isServerHealthy() );
```

along these lines we should probably think of renaming `state()` to `eval()` or something like that.